### ISSUE TITLE
Reducing the size of pypi package

### ISSUE THREAD

**Issue by rainwoodman** [manodeep/Corrfunc#102]
1. v2 is not released on pypi

2. The current pypi package is about 30MB in size. Is it really necessary? 

For comparison, the scipy source tar ball is 10 MB in size, and scipy has a lot more functionality than CorrFunc.

**Comment by rainwoodman** [comment]
Most of the 30 MB belongs to the testing data in Theory directory.

**Comment by manodeep** [comment]
Thanks for the reminder - I find the repo size too large as well and I tried in issue #51 to fix that. Unfortunately, that only reduced the repo somewhat and didn't quite get it down to acceptable levels. 

Looking at the data files again, I see that the `random_Zspace.ff` could potentially be removed and then regenerated during the tests with a fixed seed and a specific random number generator method. However, doing so does require changing/verifying the `known correct` test results. 

Do-able but currently I would rather wait until the paper is submitted; at which point I will update the PyPI repo as well (see existing issue #82). 

Thanks again for the feedback. Do you mind if I assign v2.1 as the target milestone (rather than v2.0)?

**Comment by lgarrison** [comment]
Most of the "weight" columns in the datasets are random too, and could even
be generated in-memory and never written to disk.  However, I decided not
to implement that partially because it makes comparison with other codes
more difficult.  Whatever space-saving changes we decide to implement, we
should at least keep a write-to-disk mode.

**Comment by manodeep** [comment]
@lgarrison Good point! 

One solution could be to have a submodule with the test data and the users can chose to not run tests and never download the datasets.

**Comment by rainwoodman** [comment]
Yes that's a good approach. But you'll need to find a place to store the test data.

**Comment by manodeep** [comment]
I was simply thinking of opening up a different github repo (under https://github.com/clusterfunc/ organization -- which is where some future version of Corrfunc is going to be). Do you have a different suggestion?

**Comment by rainwoodman** [comment]
Good idea. That's what Nick has set up for nbodykit. I used to use amazon
S3 -- less convenient indeed.
